### PR TITLE
more offline install fixes

### LINKED ISSUE #313: Reverse proxy on port 80 as default component
The nginx reverse proxy is currently missing in the optional components menu

### LINKED ISSUE #825: Offline install
Is openHABian install supposed to finish even when Internet access isn't available ?

At least a number of users expect it to. I myself, frankly, I don't know. @ThomDietrich ?
We have a number of reported install fails at
https://github.com/openhab/openhabian/blob/master/functions/openhab.bash#L60

While not fully analyzed yet, it all boils down to the introductory question.
If yes, we would also need to test install runs under this condition.
If yes, we would need to identify the basicmost packages we need to get openHABian+openHAB running and include them with the image:
That would at least be needed_packages() - probably only python - PLUS Java PLUS openHAB packages (current release).

### PR DESCRIPTION
**Pull request by mstormi**
Fixes #313 
Fixes #825 

Signed-off-by: Markus Storm markus.storm@gmx.net

### CODE REVIEW

File: build-image/first-boot.bash [hunk 1]
```diff
@@ -161,10 +162,11 @@ if ! cond_redirect dpkg -s "git" &>/dev/null; then
   if apt-get install --yes git &>/dev/null; then echo "OK"; else echo "FAILED"; fail_inprogress; fi
 fi
 
-if [ -d /opt/openhabian ]; then cd /opt && rm -rf /opt/openhabian; fi
+# must not remove for offline to work
+#if [ -d /opt/openhabian ]; then cd /opt && rm -rf /opt/openhabian; fi
```
Comment (holgerfriedrich) [inline build-image/first-boot.bash:166]: I suppose that removing the delete will cause `git clone` command to fail....
```
fatal: destination path '/opt/openhabian already' exists and is not an empty directory.
```

Comment (mstormi) [reply build-image/first-boot.bash:166]: right, good you spotted that.
How about this now ?

**Comment by mstormi** [comment]
> Did you test?

Unit testing only.
Will try a full install on my test box before merging.

**Comment by holgerfriedrich** [comment]
@mstormi are you going tomerge this after the tests have completed? I would build another test image later today.

**Comment by mstormi** [comment]
Successfully tested that it does not break installation with Inet available.
Tested with Inet blocked, too, found and fixed a couple of locations that offline install choked on.
Now install successfully finishes, too.
There's some tiles missing on the resulting dashboard but OH is running.

On a sidenote, @holgerfriedrich @ecdye does anyone know a HTML trick to have a tail -f like display of the web install log like in Travis rather than to reload the page every n seconds ?

**Pull request merged**